src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RemoteUserFactory.php

```diff
@@ -24,7 +24,7 @@
  */
 class RemoteUserFactory implements SecurityFactoryInterface
 {
-    public function create(ContainerBuilder $container, $id, $config, $userProvider, $defaultEntryPoint)
+    public function create(ContainerBuilder $container, string $id, array $config, ?string $userProvider, ?string $defaultEntryPoint)
```

Comment:
If `$userProvider` is null, the code breaks because its used as a reference. The null only seems to be used in \Symfony\Bundle\SecurityBundle\DependencyInjection\SecurityExtension::createAuthenticationListeners when using the RememberMeFactory. And this factory doesn't even use the userProvider. So maybe instead of null, we can just pass an empty string. Then the argument does not need to be nullable.

Reply:
Or we need to add explicit exceptions when the userProvider is null but the implementation cannot handle it.

Reply:
Thanks for catching this. I decided to not accept `null` here except for inside the `RememberMeFactory`.